src/main.ts

```diff
@@ -1,114 +1,68 @@
 import * as fs from 'fs'
 import * as cp from 'child_process'
 import * as core from '@actions/core'
-import * as toolCache from '@actions/tool-cache'
-import path from 'path'
+import * as path from 'path'
 
 async function run(): Promise<void> {
   try {
+    const localAppDataPath = process.env.LocalAppData
     // get version number from input
     const version = core.getInput('version')
 
-    // get architecture and os
-    const architecture = process.arch
+    // get os
     const os = process.platform
 
-    // map for different platform and arch
-    const extensionMap = {
-      linux: '.tar.gz',
-      darwin: '.zip',
-      win32: '.zip'
+    let windowsInstallScript = `powershell -ex AllSigned -c "Invoke-RestMethod 'https://aka.ms/install-azd.ps1' | Invoke-Expression"`
+    let linuxOrMacOSInstallScript = `curl -fsSL https://aka.ms/install-azd.sh | sudo bash`
+    if (version !== 'latest') {
+      windowsInstallScript = `powershell -ex AllSigned -c "Invoke-RestMethod 'https://aka.ms/install-azd.ps1' -OutFile 'install-azd.ps1'; powershell -ExecutionPolicy Bypass -File ./install-azd.ps1 -Version '${version}'"`
+      linuxOrMacOSInstallScript = `curl -fsSL https://aka.ms/install-azd.sh | sudo bash -s -- --version ${version}`
     }
 
-    const exeMap = {
-      linux: '',
-      darwin: '',
-      win32: '.exe'
-    }
-
-    const arm64Map = {
-      x64: 'amd64',
-      arm64: 'arm64-beta'
-    }
-
-    const platformMap = {
-      linux: 'linux',
-      darwin: 'darwin',
-      win32: 'windows'
+    core.info(`Installing azd version ${version} on ${os}.\n`)
+
+    if (os === 'win32') {
+      cp.execSync(windowsInstallScript)
+      // Add azd to PATH
+      if (localAppDataPath) {
+        const azdPath = path.join(localAppDataPath, 'Programs', 'Azure Dev CLI')
+        fs.appendFileSync(
+          process.env.GITHUB_PATH || '',
+          `${azdPath}${path.delimiter}`
+        )
+      } else {
+        core.setFailed('LocalAppData environment variable is not defined.')
```

Comment:
Just a suggestion, we could have an explicit early `return` after the `setFailed` and possibly get rid of the second `localAppDataPath` check further down below. Example:

```ts
      if (!localAppDataPath) {
        core.setFailed('LocalAppData environment variable is not defined.');
        return;
      }
      cp.execSync(windowsInstallScript)
     ...
```


Any thoughts on returning early after the `setFailed` call? Then the `localAppDataPath` check below could possibly be eliminated.